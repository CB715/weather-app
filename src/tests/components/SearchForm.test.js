import React from "react";
import { render } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("Search Form", () => {
  const validProps = {
    searchText: "Manchester",
    setSearchText: jest.fn,
    onSubmit: jest.fn(),
  };

  it("renders the component", () => {
    const { asFragment } = render(
      <SearchForm
        searchText={validProps.searchText}
        setSearchText={validProps.setSearchText}
        onSubmit={validProps.onSubmit}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
