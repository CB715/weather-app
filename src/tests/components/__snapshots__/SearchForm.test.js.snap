// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form renders the component 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      placeholder="Enter city..."
      type="text"
      value="Manchester"
    />
    <button
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
